lib/bn.js

```diff
@@ -731,6 +731,41 @@ BN.prototype.uxor = function uxor(num) {
 };
 
 
+// Not ``this`` with ``width`` bitwidth
+BN.prototype.inotn = function inotn(width) {
+  assert(typeof width === 'number' && width >= 0);
+
+  var bytesNeeded = Math.ceil(width / 26) | 0;
+  var bitsLeft = width;
+
+  // Extend the buffer with leading zeroes
+  while (this.length < bytesNeeded) {
+    this.words[this.length++] = 0;
+  }
+
+  // Handle complete words
+  for (var i = 0; i < bytesNeeded && bitsLeft >= 26; i++, bitsLeft -= 26) {
+    var tmp = this.words[i] & 0xfc000000;
+    this.words[i] = tmp | (~(this.words[i] & 0x3ffffff)) & 0x3ffffff;
+  }
+
+  // Handle the residue
+  if (bitsLeft > 0) {
```

Comment:
Either you take track of the bits left in the loop and stop when appropriate; or calculate the ending point before, or simply just leave the top part always to the residue loop.

```
@@ -736,7 +736,7 @@ BN.prototype.inotn = function inotn(width) {
   assert(typeof width === 'number' && width >= 0);
 
   var bytesNeeded = Math.ceil(width / 26) | 0;
-  var bitsLeft = width;
+  var bitsLeft = width % 26;
 
   // Extend the buffer with leading zeroes
   while (this.length < bytesNeeded) {
@@ -744,7 +744,7 @@ BN.prototype.inotn = function inotn(width) {
   }
 
   // Handle complete words
-  for (var i = 0; i < bytesNeeded && bitsLeft >= 26; i++, bitsLeft -= 26) {
+  for (var i = 0; i < bytesNeeded - 1; i++) {
     var tmp = this.words[i] & 0xfc000000;
     this.words[i] = tmp | (~(this.words[i] & 0x3ffffff)) & 0x3ffffff;
   }
```

To calculate actual end point, replace in the above:
```
+  for (var i = 0; i < bytesNeeded - ((bitsLeft > 0) ? 1 : 0); i++) {
```

Which way you prefer?